Intake and Sensors

Yumnah2 opened the pull request.

Yumnah2 committed:
Change to injest and added Sensor Override

The lower index motor is now injest and is grouped with the four base intake motors. Shuffleboard method that overrides sensors

TAKBS2412 commented on src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java:
```diff
@@ -34,6 +34,9 @@ public void end(boolean interrupted) {
 	@Override
 	public boolean isFinished() {
-		return intakeSubsystem.getFeederSensor();
+		if (!intakeSubsystem.getSensorOverride()) {
```
This can be rewritten to use ||.

eddiert1239 replied:
yeah you're right

Yumnah2 committed:
addressed comment

rephrased return in isFinished in AllInCommand

TAKBS2412 commented on src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java:
```diff
@@ -34,6 +34,6 @@ public void end(boolean interrupted) {
 	@Override
 	public boolean isFinished() {
-		return intakeSubsystem.getFeederSensor();
+		return !intakeSubsystem.getSensorOverride() || intakeSubsystem.getFeederSensor();
```
I think this is different from the older version of the logic, if sensor override is true but feeder sensor is false, the old version would return true but the new version returns false.

eddiert1239 replied:
yeah you're right

Yumnah2 committed:
fixed this thing again

bobtseattle commented on src/main/java/frc/team2412/robot/commands/intake/AllInCommand.java:
```diff
@@ -20,7 +20,7 @@ public void initialize() {
 	@Override
 	public void execute() {
-		if (intakeSubsystem.getIndexSensor()) {
+		if (intakeSubsystem.getIndexSensor() && !intakeSubsystem.getSensorOverride()) {
```
To make the logic more clear, is it better to override the sensor here or in the getSensorOverride method?
Or with a parameter to getSensorOverride that overrides the sensor?

bobtseattle reviewed (commented):
I am debating if there is a better way to override the sensor value so we don't have to implement extra logic in this and other commands(if ever needed)?

Yumnah2 committed:
Add intake reject and changed sensor override

Intake reject command
Dpad down for that command
Override logic is within sensor methods

eddiert1239 commented:
> I am debating if there is a better way to override the sensor value so we don't have to implement extra logic in this and other commands(if ever needed)?

Both the getFeederSensor() and getIndexSensor() methods now return sensor value if not overriding, and false if we are overriding. This means no extra logic in commands.

Yumnah2 committed:
Rename get sensor methods to sensor Has Note

Yumnah2 committed:
made sensors work properly

return false means there is a note, so we fixed that in our code

bobtseattle reviewed (commented):
LGTM.  Would be nice if the sensor method would return true to make it more intuitive, but we can go with the ! until we get confused.

eddiert1239 committed:
logging sensors is correct values

eddiert1239 committed:
Update IntakeSubsystem.java

eddiert1239 committed:
works now

eddiert1239 committed:
hopefully now

KangarooKoala commented on src/main/java/frc/team2412/robot/subsystems/IntakeSubsystem.java:
```diff
@@ -214,12 +223,16 @@ public void feederShoot() {
-	public boolean getIndexSensor() {
-		return indexSensor.get();
+	public boolean indexSensorHasNote() {
+		return !indexSensor.get() && !getSensorOverride();
```
Why is `indexSensor.get()` inverted now? (This also applies to `feederSensorHasNote()`)

eddiert1239 replied:
.get() returns true as a default, and if it senses something it returns false

KangarooKoala reviewed (commented):
Looks pretty good to me, except the one question (and also the correct spelling is ingest, not injest).

eddiert1239 committed:
injest = ingest
